tests/wp-unit/include/Core/Posts/Modules/DeletePostsByPostTypeModuleTest.php

```diff
@@ -0,0 +1,315 @@
+		$this->assertEquals( 10, count( $published_posts ) );
+
+		$delete_options = array(
+			'selected_types'  => array( 'custom', 'post' ),
+			'limit_to'     => -1,
+			'restrict'     => false,
+			'force_delete' => true,
+		);
+
+		$posts_deleted = $this->module->delete( $delete_options );
+		$this->assertEquals( 20, $posts_deleted );
+
+		$trash_posts = $this->get_posts_by_status( 'trash' );
+		$this->assertEquals( 0, count( $trash_posts ) );
+
+	}
+
+	/**
+	 * Add tests to deleting posts that are older than x days.
+	 */
+	public function test_delete_posts_that_are_older_than_x_days() {
+		$date = date( 'Y-m-d H:i:s', strtotime( '-5 day' ) );
+
+		$published_posts = $this->factory->post->create_many( 10, array(
+			'post_type'   => 'post',
+			'post_date'   => $date,
+		) );
+
+		$this->assertEquals( 10, count( $published_posts ) );
+
+		$delete_options = array(
+			'selected_types'  => array( 'post' ),
+			'limit_to'     => -1,
+			'restrict'     => false,
+			'force_delete' => false,
+			'date_op'      => 'before',
+			'days'         => '3',
+		);
+
+		$posts_deleted = $this->module->delete( $delete_options );
+		$this->assertEquals( 10, $posts_deleted );
+
+		$published_posts = $this->get_posts_by_post_type();
+		$this->assertEquals( 0, count( $published_posts ) );
+
+	}
+
+	/**
+	 * Add tests to deleting posts that past x days.
+	 */
+	public function test_delete_posts_that_are_past_x_days() {
+		$date = date( 'Y-m-d H:i:s', strtotime( '-3 day' ) );
+
+		$published_posts = $this->factory->post->create_many( 10, array(
+			'post_type'   => 'post',
+			'post_date'   => $date,
+		) );
+
+		$this->assertEquals( 10, count( $published_posts ) );
+
+		$delete_options = array(
+			'selected_types'  => array( 'post' ),
+			'limit_to'     => -1,
+			'restrict'     => false,
+			'force_delete' => false,
+			'date_op'      => 'after',
+			'days'         => '5',
+		);
+
+		$posts_deleted = $this->module->delete( $delete_options );
+		$this->assertEquals( 10, $posts_deleted );
+
+		$published_posts = $this->get_posts_by_post_type();
+		$this->assertEquals( 0, count( $published_posts ) );
+
+	}
+
+	/**
+	 * Add tests to deleting posts by batches.
+	 */
+	public function test_delete_posts_by_batchs() {
+
+		$published_posts = $this->factory->post->create_many( 100, array(
+			'post_type'   => 'post',
+		) );
+
+		$this->assertEquals( 100, count( $published_posts ) );
+
+		$delete_options = array(
+			'selected_types'  => array( 'post' ),
+			'limit_to'     => 50,
+			'restrict'     => false,
+			'force_delete' => false,
+		);
+
+		$posts_deleted = $this->module->delete( $delete_options );
+		$this->assertEquals( 50, $posts_deleted );
+
+		$published_posts = $this->get_posts_by_post_type();
+		$this->assertEquals( 50, count( $published_posts ) );
+
+	}
+
+	/**
+	 * Helper function.
+	 * create posts by custom post type
+	 */
+	public function create_posts_by_custom_post_type( $post_type = 'custom', $count = 10 ) {
```

Comment:
Make the following changes for this helper functions

- Move the call to `register_post_type` outside of this helper function.
- Combine `create_posts_by_custom_post_type` and `create_post_by_custom_post_type` into a single function.